In [53]:
import pandas as pd
import numpy as np

In [5]:
dataMod = pd.read_csv('resultsMod.csv')
dataOr = pd.read_csv('resultsOriginal.csv')

In [64]:
dataMod

targetsDetected   destroyed   whereDestroyed.x   missionSuccess  \
0                    0           0                 39                0   
1                    1           0                 39                0   
2                    2           0                 39                1   
3                    0           0                 39                0   
4                    0           0                 39                0   
5                    1           0                 39                0   
6                    2           0                 39                1   
7                    0           0                 39                0   
8                    0           0                 39                0   
9                    0           0                 39                0   
10                   1           1                 38                0   
11                   1           1                 34                0   
12                   0           1                 13                0   
13                   0           0                 39                0   
14                   1           0                 39                0   
15                   2           0                 39                1   
16                   1           0                 39                0   
17                   3           0                 39                1   
18                   1           0                 39                0   
19                   1           0                 39                0   
20                   1           1                 28                0   
21                   0           0                 39                0   
22                   0           0                 39                0   
23                   0           0                 39                0   
24                   0           0                 39                0   
25                   1           0                 39                0   
26                   1           0                 39                0   
27                   0           0                 39                0   
28                   0           0                 39                0   
29                   1           0                 39                0   
...                ...         ...                ...              ...   
99970                3           0                 39                1   
99971                1           1                 39                0   
99972                1           1                 24                0   
99973                0           0                 39                0   
99974                1           0                 39                0   
99975                0           0                 39                0   
99976                0           0                 39                0   
99977                0           0                 39                0   
99978                1           1                 18                0   
99979                0           0                 39                0   
99980                0           1                  5                0   
99981                1           1                 29                0   
99982                0           0                 39                0   
99983                0           0                 39                0   
99984                0           0                 39                0   
99985                0           0                 39                0   
99986                1           0                 39                0   
99987                1           1                 36                0   
99988                1           0                 39                0   
99989                0           0                 39                0   
99990                1           0                 39                0   
99991                1           0                 39                0   
99992      

In [65]:
dataOr

targetsDetected   destroyed   whereDestroyed.x   missionSuccess  \
0                    1           0                 39                0   
1                    1           0                 39                0   
2                    1           0                 39                0   
3                    0           0                 39                0   
4                    1           0                 39                0   
5                    0           1                  9                0   
6                    2           0                 39                1   
7                    2           0                 39                1   
8                    0           0                 39                0   
9                    1           0                 39                0   
10                   1           0                 39                0   
11                   2           0                 39                1   
12                   1           0                 39                0   
13                   0           0                 39                0   
14                   1           0                 39                0   
15                   1           0                 39                0   
16                   0           0                 39                0   
17                   2           0                 39                1   
18                   1           0                 39                0   
19                   2           0                 39                1   
20                   1           0                 39                0   
21                   0           0                 39                0   
22                   1           0                 39                0   
23                   3           0                 39                1   
24                   2           0                 39                1   
25                   0           1                 18                0   
26                   1           0                 39                0   
27                   2           0                 39                1   
28                   1           0                 39                0   
29                   0           0                 39                0   
...                ...         ...                ...              ...   
99970                1           0                 39                0   
99971                2           0                 39                1   
99972                0           0                 39                0   
99973                2           0                 39                1   
99974                1           0                 39                0   
99975                1           0                 39                0   
99976                3           0                 39                1   
99977                0           0                 39                0   
99978                1           0                 39                0   
99979                0           0                 39                0   
99980                1           0                 39                0   
99981                0           0                 39                0   
99982                0           0                 39                0   
99983                2           0                 39                1   
99984                0           0                 39                0   
99985                0           0                 39                0   
99986                0           0                 39                0   
99987                0           0                 39                0   
99988                1           0                 39                0   
99989                1           0                 39                0   
99990                1           0                 39                0   
99991                2           0                 39                1   
99992      

# original Stats

###  Success rate

In [93]:
successCounts = np.unique(np.array(dataOr.iloc[:,3:4]),return_counts=True)
print( "Success rate: ", successCounts[1][1]/(successCounts[1][1]+successCounts[1][0]))

Success rate:  0.26827


### Destroyed Rate

In [94]:
destroyedCounts = np.unique(np.array(dataOr.iloc[:,1:2]),return_counts=True)
print( "Destroyed rate: ", destroyedCounts[1][1]/(destroyedCounts[1][1]+destroyedCounts[1][0]))

Destroyed rate:  0.02683


### Decision TimeAvg Stats

In [89]:
dataOr.iloc[:,4:5].describe()

decisionTimeAvg
count     100000.000000
mean           0.017355
std            0.024615
min           -0.818802
25%            0.010600
50%            0.013845
75%            0.019480
max            1.474530

### Decision Time variance Stats

In [90]:
dataOr.iloc[:,5:6].describe()

decisionTimeVar
count     100000.000000
mean           0.022259
std            0.722692
min            0.000041
25%            0.000159
50%            0.000278
75%            0.000589
max           84.201700

# Stats for random values between 0-5

### Success Rate

In [96]:
modSuccessCounts = np.unique(np.array(dataMod.iloc[:,3:4]),return_counts=True)
print( "Success rate: ", modSuccessCounts[1][1]/(modSuccessCounts[1][1]+modSuccessCounts[1][0]))

Success rate:  0.12586


### Destroyed Rate

In [97]:
modDestroyedCounts = np.unique(np.array(dataMod.iloc[:,1:2]),return_counts=True)
print( "Destroyed rate: ", modDestroyedCounts[1][1]/(modDestroyedCounts[1][1]+modDestroyedCounts[1][0]))

Destroyed rate:  0.16095


### Decision TimeAvg Stats

In [98]:
dataMod.iloc[:,4:5].describe()

decisionTimeAvg
count     100000.000000
mean           0.017259
std            0.032467
min           -1.207580
25%            0.011458
50%            0.014742
75%            0.019184
max            3.546280

### Decision Time variance Stats

In [99]:
dataMod.iloc[:,5:6].describe()

decisionTimeVar
count     100000.000000
mean           0.032374
std            1.150742
min            0.000035
25%            0.000195
50%            0.000349
75%            0.000565
max          162.706000

### Observations

##### These results are based on 100,000 simulations of the dartsim adaption manager. 
##### For random simulation of random latencies, the value was limited to 5 periods. The defualt value for latency is 1 period.
##### The most visible differences between the two sets of simulations are the success and destroyed rate. Having random values for latencies negatively affects the performance of the adaption manager.

##### Latency = 1 vs. Latency = (random)
##### Success rate = 0.26827 vs. 0.12586
##### Destroyed rate = 0.02683 vs. 0.16095 
